tiledb/sm/serialization/query.cc

```diff
@@ -923,6 +908,54 @@ Status query_deserialize(
   STATS_FUNC_OUT(serialization_query_deserialize);
 }
 
+Status query_deserialize(
+    const Buffer& serialized_buffer,
+    SerializationType serialize_type,
+    bool clientside,
+    CopyState* copy_state,
+    Query* query) {
+  // Create an original, serialized copy of the 'query' that we will revert
+  // to if we are unable to deserialize 'serialized_buffer'.
+  BufferList original_bufferlist;
+  Status st =
+      query_serialize(query, serialize_type, clientside, &original_bufferlist);
```

Comment:
I would rewrite all of these without the explicit `st` local variable, e.g. `RETURN_NOT_OK(query_serialize(...));`

Reply:
Done